libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Classes.hs

```diff
@@ -932,3 +923,159 @@ partitionWord64 small msgs n total =
 partitionNatural :: Natural -> [String] -> Int -> Natural -> Gen [Natural]
 partitionNatural small msgs n total =
   map fromIntegral <$> integerPartition msgs "Natural" (fromIntegral small) n (fromIntegral total)
+
+-- | Translate (s,cond,n), into a Size which
+--   specifies the Int range on which the OrdCond is True.
+--   The triple (s, EQL, 2) denotes s = 2
+--              (s, LTH, 7) denotes s < 7
+--              (s, GTH, 5) denotes s > 5 ...
+ordCondToSize :: Adds a => (String, OrdCond, a) -> Size
+ordCondToSize (label, cond, n) = case cond of
+  EQL -> SzExact $ toI n
+  LTH -> SzMost $ toI $ decrease [unlines ["ordCondToSize failed for", label, show n]] n
```

Comment:
decrease n = pair (i-1) (j-1)  where (i,j) = unpair n


EQL -> SzExact $ toI n
LTH -> SzMost $ decrease [unlines ["ordCondToSize failed for", label, show n]] (toI n)
 GTH -> SzLeast $ increase (toI n)